CMakeLists.txt

```diff
@@ -1,17 +1,27 @@
-CMAKE_MINIMUM_REQUIRED(VERSION 2.8.12 FATAL_ERROR)
+CMAKE_MINIMUM_REQUIRED(VERSION 3.0 FATAL_ERROR)
 
 INCLUDE(GNUInstallDirs)
```

Comment:
While you're in here it'd be good to move this include to after `PROJECT` since the `GNUInstallDirs` module uses platform information initialized by `PROJECT` to determine the default lib directory:
```
CMake Warning (dev) at /opt/cmake/3.27.7/share/cmake-3.27/Modules/GNUInstallDirs.cmake:243 (message):
  Unable to determine default CMAKE_INSTALL_LIBDIR directory because no
  target architecture is known.  Please enable at least one language before
  including GNUInstallDirs.
Call Stack (most recent call first):
  CMakeLists.txt:3 (INCLUDE)
This warning is for project developers.  Use -Wno-dev to suppress it.
```